### ISSUE TITLE
Improved image colour filtering/similarity

### ISSUE THREAD

**Issue by jamieparkinson** [wellcomecollection/platform#5703]
Following on from https://github.com/wellcomecollection/data-science/issues/139, it's now very straightforward for us to use bona fide vectors to represent colours and search for their neighbours. From this, we'll get to delete a decent chunk of code and also to quite significantly improve the performance of colour filtering/similarity.

- [x] Update the [palette inferrer](https://github.com/wellcomecollection/catalogue-pipeline/tree/bc9ce4802f7f655721cbea88b4e67c78aa6ff31e/pipeline/inferrer/palette_inferrer) app in the catalogue pipeline to return vectors.

  Most of this will be copying and pasting from the [data-science repo](https://github.com/wellcomecollection/data-science/blob/main/knn-colours/pipeline/src/embedder.py) and deleting the existing logic. The scaffolding for getting the image from the URL should stay the same, and I imagine it will be easiest to keep the batching stuff too.

  The inferrer currently returns some of its parameters as well as the palette strings. These are used by the catalogue API to make sure it can construct colour vectors in the correct basis. It looks like we'll want to do the same with `alpha` and `n_bins` [here](https://github.com/wellcomecollection/catalogue-api/blob/2a42c55e7c492f5b04b65cf0c6252c881a613ae4/search/src/main/scala/weco/api/search/models/QueryConfig.scala#L22-L32) - @harrisonpim will be able to confirm.

  There is a precedent in the `feature_inferrer` for returning vectors in [base64](https://github.com/wellcomecollection/catalogue-pipeline/blob/af4da934ad2956c3c6cb6df305c70b6d346fdd07/pipeline/inferrer/feature_inferrer/app/main.py#L65) to save quite a lot of network bandwidth - possibly best to replicate that here just so we can copy and paste more!

- [x] Make sure the [inference manager](https://github.com/wellcomecollection/catalogue-pipeline/blob/996a06121643e6b714221029bb47d4add71af077/pipeline/inferrer/inference_manager/src/main/scala/weco/pipeline/inference_manager/models/PaletteInferrerResponse.scala) knows what to expect from the palette inferrer and how to [deal with it](https://github.com/wellcomecollection/catalogue-pipeline/blob/996a06121643e6b714221029bb47d4add71af077/pipeline/inferrer/inference_manager/src/main/scala/weco/pipeline/inference_manager/adapters/PaletteInferrerAdapter.scala#L1-L48). You can see how we deal with the feature vectors already, eg [here](https://github.com/wellcomecollection/catalogue-pipeline/blob/996a06121643e6b714221029bb47d4add71af077/pipeline/inferrer/inference_manager/src/main/scala/weco/pipeline/inference_manager/models/PaletteInferrerResponse.scala) and [here](https://github.com/wellcomecollection/catalogue-pipeline/blob/97e4a43ad57b26d7e933c1f2708716a46eedd3ff/pipeline/inferrer/inference_manager/src/main/scala/weco/pipeline/inference_manager/adapters/FeatureVectorInferrerAdapter.scala#L41-L74). The linked bits of code aren't exhaustive, but the tests and the compilation will tell you what else to do!

- [x] Update the [index mapping](https://github.com/wellcomecollection/catalogue-pipeline/blob/ddef0bd692caa41f426b0a01f2b18aef44bdbaa4/index_config/mappings.images_indexed.2023-05-26.json#L89-L91) to expect vectors.

- [ ] Do a reindex! My tip here would be that not everything needs to be done in order to test whether the inferrer works and whether the right stuff is in the index (in fact I'd recommend not starting with everything, just send through an individual image. The reindexer script knows how to do this). 

- [ ] Update the [catalogue API](https://github.com/wellcomecollection/catalogue-api/blob/7901becce4c7e4112361126ea7d8502be3054513/search/src/main/scala/weco/api/search/elasticsearch/ColorQuery.scala) be parametrised by the new query config (as mentioned in the first task) and then to make the correct queries: those will look something like doing a kNN query in the `apply` method of `ColorQuery` and changing the absolute chaos that is `getColorsSignature` to look something like [this much simpler implementation](https://github.com/wellcomecollection/data-science/blob/main/knn-colours/webapp/helpers/colours.tsx#L29-L44).

  Basically what this class does is, given the parameters of the colour palette vector model, creates a vector for an arbitrary colour and returns a query for it. You will see that we convert the hex codes to HSV and deal with HSV values; this was probably a bit misguided and you can remove all of it and deal only with RGB now.
